pkg/sql/internal.go

```diff
@@ -879,6 +884,11 @@ func (ie *InternalExecutor) execInternal(
 	timeReceived := timeutil.Now()
 	parseStart := timeReceived
 	parsed, err := parser.ParseOne(stmt)
+	if tree.CanModifySchema(parsed.AST) && txn != nil {
+		if ie.extraTxnState == nil || ie.extraTxnState.descCollection == nil {
```

Comment:
I think we need a method on `ie` to detect whether it has a bound transaction or not. All of this nil checking and differentiated nil checking is hard to read. Also, combine the conditions logically as opposed to nesting the if statements.

Reply:
Done.